src/main/java/org/junit/internal/runners/statements/FailOnTimeout.java

```diff
@@ -121,28 +121,21 @@ public void evaluate() throws Throwable {
         CallableStatement callable = new CallableStatement();
         FutureTask<Throwable> task = new FutureTask<Throwable>(callable);
         ThreadGroup threadGroup = new ThreadGroup("FailOnTimeoutGroup");
-        Thread thread = new Thread(threadGroup, task, "Time-limited test");
-        try {
-            thread.setDaemon(true);
-            thread.start();
-            callable.awaitStarted();
-            Throwable throwable = getResult(task, thread);
-            if (throwable != null) {
-                throw throwable;
-            }
-        } finally {
+        if (!threadGroup.isDaemon()) {
             try {
-                thread.join(1);
-            } catch (InterruptedException e) {
-                Thread.currentThread().interrupt();
-            }
-            try {
-                threadGroup.destroy();
-            } catch (IllegalThreadStateException e) {
-                // If a thread from the group is still alive, the ThreadGroup cannot be destroyed.
+                threadGroup.setDaemon(true);
+            } catch (SecurityException e) {
                 // Swallow the exception to keep the same behavior prior to this change.
```

Comment:
The term "this change" is a little bit unspecific now. Can you change it to
```suggestion
                // Swallow the exception to keep the same behavior as in JUnit 4.12.
```

Reply:
How

Reply:
@shivam11-u I'm not quite sure what your question is, but if you are wondering how this change "keeps the same behavior" code here is catching `SecurityException` it's because it's possible that JUnit is running in an environment